README.md

```diff
@@ -1 +1,24 @@
-# gridlington-datahub
\ No newline at end of file
+# gridlington-datahub
+
+## Development
+
+### Installation
+
+1. Create a virtual environment with python 3.10 or higher: `python -m venv .venv`
+2. Activate it: `source .venv/bin/activate`
+3. Install development requirements in the virtual environment: `pip install -r requirements-dev.txt`.
+4. Install pre-commit hooks: `pre-commit install`. QA can be checked with `pre-commit run --all-files` and will automatically check files before commiting them to git history.
+5. Run tests: `pytest`. This will create a coverage report inside `htmlcov/`.
+
+### Dependencies
+Dependencies are managed using the [`pip-tools`] tool chain. Unpinned dependencies are specified in `pyproject.toml`. Pinned versions are then produced with: `pip-compile`.
+
+To add/remove packages edit `pyproject.toml` and run the above command. To upgrade all existing dependencies run: `pip-compile --upgrade`.
+
+Dependencies for developers are listed separately as optional, with the pinned versions being saved to `requirements-dev.txt` instead. `pip-tools` can also manage these dependencies by adding extra arguments, e.g.: `pip-compile --extra dev -o requirements-dev.txt`.
+
+When dependencies are upgraded, both `requirements.txt` and `requirements-dev.txt` should be regenerated so that they are compatible with each other and then synced with your virtual environment with: `pip-sync requirements-dev.txt requirements.txt`.
+
+Dependabot is configured to update dependencies as new versions become available.
```

Comment:
GitHub claims that [dependant supports `pip-tools`](https://github.blog/changelog/2021-06-10-dependabot-updated-pip-pip-tools-and-pipenv-support/).

I believe it will open a PR and edit the `requirements.txt` files and respect any restrictions to versions we put in the `pyproject.toml`. At least that's how I think it _should_ work.

> I wonder whether it might be worth adding to that section instructions on how to ensure that a package that should not be upgraded for compatibility reasons is properly pinned?

I have two thoughts on this:
1. `pip-tools` should take care of any compatibility issues between required packages. So if dependant _truly_ supports `pip-tools` that should be a non-issue.
2. Restricting packages from upgrading through our own choice (an update makes backwards-incompatible changes) is something we have control over, will add some instructions.